src/Gui/CommandView.cpp

```diff
@@ -3614,6 +3614,70 @@ Action * StdCmdSelBoundingBox::createAction()
     return pcAction;
 }
 
+//===========================================================================
+// Std_StoreWorkingView
+//===========================================================================
+DEF_STD_CMD_A(StdStoreWorkingView)
+
+StdStoreWorkingView::StdStoreWorkingView()
+  : Command("Std_StoreWorkingView")
+{
+    sGroup        = "Standard-View";
+    sMenuText     = QT_TR_NOOP("Store working view");
+    sToolTipText  = QT_TR_NOOP("Store a document-specific temporary working view");
+    sStatusTip    = QT_TR_NOOP("Store a document-specific temporary working view");
+    sWhatsThis    = "Std_StoreWorkingView";
+    sAccel        = "Shift+End";
+    eType         = NoTransaction;
+}
+
+void StdStoreWorkingView::activated(int iMsg)
+{
+    Q_UNUSED(iMsg);
+    if (isActive()) {
+        auto view = static_cast<Gui::View3DInventor*>(Gui::getMainWindow()->activeWindow());

```

Comment:
The static_cast relies on the correct implementation of isActive(). To be more robust you should use dynamic_cast and check the returned pointer.

```
void StdStoreWorkingView::activated(int iMsg)
{
    Q_UNUSED(iMsg);
    if (auto view = dynamic_cast<Gui::View3DInventor*>(Gui::getMainWindow()->activeWindow());) {
        view->getViewer()->saveHomePosition();
    }
}
```